src/Io/StreamingServer.php

```diff
@@ -96,18 +98,20 @@ final class StreamingServer extends EventEmitter
      *
      * @param LoopInterface $loop
      * @param callable $requestHandler
+     * @param float $idleConnectTimeout
      * @see self::listen()
      */
-    public function __construct(LoopInterface $loop, $requestHandler)
+    public function __construct(LoopInterface $loop, $requestHandler, $idleConnectTimeout = InactiveConnectionTimeoutMiddleware::DEFAULT_TIMEOUT)
     {
         if (!\is_callable($requestHandler)) {
             throw new \InvalidArgumentException('Invalid request handler given');
         }
 
         $this->loop = $loop;
+        $this->idleConnectionTimeout = $idleConnectTimeout;
 
         $this->callback = $requestHandler;
-        $this->parser = new RequestHeaderParser();
+        $this->parser = new RequestHeaderParser($this->loop, $this->idleConnectionTimeout);
```

Comment:
As in also make it possible to inject a different header parser? We could do that so we can have a C based extension for parser to speed it up?

Reply:
KISS: The `StreamingServer` is marked `@internal`, as is the `RequestHeaderParser`. This means we can easily inject the `RequestHeaderParser` to the `StreamingServer` inside the `Server` class. Constructor may have to be changed, but doesn't affect BC, because the public `Server` interface is not affected (at least not as part of this change).

Reply:
Yes, I'm fully aware, I'm also thinking ahead. But maybe, if we want to go there, we should ship that C based extension to speed it up

Reply:
I'm not sure this is currently our hot path, but if we have evidence an optional extension will speed things up, I'm all for it 👍 I think we're on the same page here and should probably leave this up for a follow-up PR?

In either case I think your changes are a welcome improvement as is and injecting the `RequestHeaderParser` into the `StreamingServer` can help simplify the dependency injection both now and longer-term 👍

Reply:
> I'm not sure this is currently our hot path, but if we have evidence an optional extension will speed things up, I'm all for it 👍 I think we're on the same page here and should probably leave this up for a follow-up PR?

Of course, that would be a follow-up PR. And we have tons of other things to do.

> In either case I think your changes are a welcome improvement as is and injecting the `RequestHeaderParser` into the `StreamingServer` can help simplify the dependency injection both now and longer-term 👍

Sounds good! 🚢 🇮🇹 !

Reply:
It took some work, but I think it's worth the refactoring :heart: